### PR TITLE
Update gradle and skip unnecessary swing updates

### LINKED ISSUE #430: search lag: the awt event queue refuses to stop until it finishes all render requests
Seems if a search yields a ton of results, the awt event queue will render all those results, and time is wasted on repeatedly expanding the ui, setting text, etc. (per jdk mission control) maybe we can choose to only show top 1000 results if we find too many.

### PR DESCRIPTION
**Pull request by liach**
update proguard to support java 17. Fixes #430

Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

**Comment by liach** [comment]
The underlying problem is still swing itself: while this allows searches like `c` for classes to be much faster and not lag for typing, when you start dragging the search bar, swing will still lag just because it is slow at handling such enormous lists :disappointed: but I guess this patch does what we want for the most part.

**Comment by liach** [comment]
Before, when enigma is stalling, the jfr's thread stack trace for the AWT event queue always has this `"cuchaz.enigma.gui.dialog.SearchDialog.lambda$updateList$8 line: 213 "` line, which is why I suspected that would be the issue.

File: enigma-swing/src/main/java/cuchaz/enigma/gui/dialog/SearchDialog.java [hunk 1]
```diff
@@ -207,7 +208,31 @@ private void updateList() {
 		this.classListModel = classListModel;
 		classList.setModel(classListModel);
 
-		currentSearch = su.asyncSearch(searchField.getText(), (idx, e) -> SwingUtilities.invokeLater(() -> classListModel.insertElementAt(e, idx)));
+		// handle these search result like minecraft scheduled tasks to prevent
+		// flooding swing buttons inputs etc with tons of (possibly outdated) invocations
+		record Order(int idx, SearchEntryImpl e) {}
```
Comment (2xsaiko) [inline enigma-swing/src/main/java/cuchaz/enigma/gui/dialog/SearchDialog.java:213]: You can declare records in methods? Whoaaaa

**Review by 2xsaiko — APPROVED** [review]
I see, so that was the problem. Very nice! Strange that it doesn't happen to me.

**Comment by liach** [comment]
yeah, somehow this lag doesn't immediately occur when opening enigma, but appears after opening a few classes (preferably hard to decompile ones) and do a few searches. dunno what's the mechanism of that, though.

File: enigma-swing/src/main/java/cuchaz/enigma/gui/dialog/SearchDialog.java (hunk 1 above)
Comment (liach) [reply enigma-swing/src/main/java/cuchaz/enigma/gui/dialog/SearchDialog.java:213]: yeah, you can actually declare local classes way before records are a thing and have name like SearchDialog$1Order

**Comment by liach** [comment]
@modmuss50 If this looks good, mind merging it? This can be quite helpful when performing method searches

**Comment by 2xsaiko** [comment]
Thanks!

**Pull request merged**